From 141f9627bb1ba36dcfba166d8823daa541416432 Mon Sep 17 00:00:00 2001
From: Dylan Van Assche <me@dylanvanassche.be>
Date: Thu, 30 Mar 2023 19:55:02 +0200
Subject: [PATCH 4/7] libqmi-glib,endpoint: handle QMUX and QRTR messages

Handle QMUX and QRTR messages in QMI Message:

- Allow QMI messages with both QMUX and QRTR markers
---
 src/libqmi-glib/qmi-endpoint.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/libqmi-glib/qmi-endpoint.c b/src/libqmi-glib/qmi-endpoint.c
index 4743e253..97503b29 100644
--- a/src/libqmi-glib/qmi-endpoint.c
+++ b/src/libqmi-glib/qmi-endpoint.c
@@ -67,7 +67,8 @@ qmi_endpoint_parse_buffer (QmiEndpoint        *self,
          * If we broke framing, an error should be reported and the device
          * should get closed */
         if (self->priv->buffer->len > 0 &&
-            self->priv->buffer->data[0] != QMI_MESSAGE_QMUX_MARKER) {
+            self->priv->buffer->data[0] != QMI_MESSAGE_QMUX_MARKER &&
+            self->priv->buffer->data[0] != QMI_MESSAGE_QRTR_MARKER) {
             g_set_error (error,
                          QMI_PROTOCOL_ERROR,
                          QMI_PROTOCOL_ERROR_MALFORMED_MESSAGE,
-- 
2.39.2

